### PR TITLE
Replace Report_AddComment with AddComment (Part 2)

### PR DESCRIPTION
**Pull request by marcaaron**
### Details

This PR:

- Remove the `reportComment` Pusher subscriptions (`updateReportWithNewAction()` remains and is used in the Urban Airship notification as I'm not sure how we are handling "onyx updates" there yet)
- Updates `DeprecatedAPI.Report_AddComment()` to `API.write('AddComment')`

To minimize changes I decided to keep the "attachment" and "comment" flows together for now.

Test in connection with: https://github.com/Expensify/Web-Expensify/pull/33951

### Fixed Issues (Related to
https://github.com/Expensify/Expensify/issues/211241

### Tests 

Same as QA but with an additional test to verify that users blocked by Concierge will see an error message inline.

**Test Blocked from Concierge**

1. Log in with a test user account
2. Begin chatting with Concierge
3. Sign into OldDot with your local expensify.com account
4. Block the test user via Concierge

5. Send a message from the user account to Concierge DM
6. Verify an inline message appears and the chat composer is disabled

### QA Steps

**Testing a report comment is sent in realtime**

1. Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
8. Send messages back and forth from each account to the other in a DM chat
9. Verify they appear in realtime

**Testing a report comment browser notification appears when the chat is not active (web)**

1. Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
2. Send message from Account A to Account B while Account B is focused on another chat
3. Verify a browser notification appears for Account B

**Testing a report comment is queued while offline and sent when online again**

1. Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
2. Send message from Account A to Account B while Account A is "offline"
3. Verify Account A sees the message in a grey "pending" state
10. Bring Account A back online
11. Verify the message darkens to indicate it was created.
12. Verify Account B can also see the message now

**Add an attachment**

1. Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
2. Add an attachment to the chat between these two users.
3. Verify the attachment appears in realtime for the other user (**Note:** it may take some time for it to upload - this is normal).

**Add an attachment (offline)**

1. Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
2. Add an attachment to the chat between these two users while offline.
3. Verify the attachment appears greyed to indicate that it is pending.
4. Bring the user online.
5. Verify the attachment darkens to indicate it is created
3. Verify the attachment appears in realtime for the other user

**Add attachment with text**

1. Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
2. Add an attachment to the chat between these two users while there is text in the composer.
3. Verify the attachment and text appears in realtime for the other user.

- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### CODE REVIEW

File: src/libs/actions/Report.js
Comment (Luke9389) [inline src/libs/actions/Report.js:899]: can we put `optimisticReportActionIDs` in here now rather than generate it and push it in after?

**Review by Luke9389 — COMMENTED** [review]
Beautiful. Just nabs and clarifying questions

File: src/components/InlineSystemMessage.js
Comment (marcaaron) [reply src/components/InlineSystemMessage.js:16]: I think yes? We can design the component however we want. In this case, if an undefined `message` is passed here we will default to empty string and then return `null` because `props.message.length === 0`.

File: src/libs/actions/Report.js
Comment (marcaaron) [reply src/libs/actions/Report.js:899]: We could. I left it as is because I think it makes an easier diff for someone to review or reference later when they check out this PR in the future vs. moving things around for no real benefit.

File: src/libs/actions/Report.js
Comment (Luke9389) [reply src/libs/actions/Report.js]: Thanks!

**Comment by marcaaron** [comment]
This is still on HOLD, but I think ready for some reviews.

**Comment by arosiclair** [comment]
Looks pretty good to me!

**Review by arosiclair — DISMISSED** [review]

**Comment by marcaaron** [comment]
This is off HOLD now and ready for final reviews. I have mainly been testing on web so far. Going to do a run through on native.

**Comment by marcaaron** [comment]
Putting a HOLD on this one while we work out some things in https://github.com/Expensify/Web-Expensify/pull/34086

**Comment by marcaaron** [comment]
This is still on hold pending the deploy of https://github.com/Expensify/Web-Expensify/pull/34086 

However, we will need to merge this relatively quickly one that code hits production so I think we should start getting some reviews in.

**Comment by marcaaron** [comment]
Taking this off hold as the linked Web-Expensify PR is on staging.

**Comment by marcaaron** [comment]
We should only merge this when it's on production though.

**Comment by marcaaron** [comment]
Testing well for me on all platforms.

**Comment by marcaaron** [comment]
Just a reminder, this very urgently needs to be merged and deployed as soon as the Web-Expensify PR goes to prod. For that reason I am also CP-ing to staging.

**Review by yuwenmemon — APPROVED** [review]
LGTM! 🎉

**Review by arosiclair — APPROVED** [review]
Looks solid 👍🏽

**Review by PauloGasparSv — APPROVED** [review]
LGTM!
Merging it already since it has to go fast

File: src/libs/actions/Report.js
Comment (bondydaa) [inline src/libs/actions/Report.js:964]: I think here I might have preferred a separate variable like 
```js
const command = isAttachment ? 'AddAttachment' : 'AddComment';
API.write(command, parameters, ...);
```

or maybe even getting rid of `isAttachment` and renaming that `command` at the top of the method and instead forking logic if command is `AddAttachment` or `AddComment` though that might end up being way too verbose 🤷

Comment (bondydaa) [reply src/libs/actions/Report.js:964]: eh i guess looking back we use `isAttachment` inside of the onxyData and a few other places so maybe this is the best route the way it's written.

File: tests/README.md
Comment (bondydaa) [inline tests/README.md:72]: shouldn't this be `isLoading`?

File: src/libs/DateUtils.js
Comment (marcaaron) [reply src/libs/DateUtils.js:146]: That could work. I'm not sure why we chose every 5 minutes or whether it would be easier to check if the timezone has changed. I think there is some kind of cost associated with trying to "guess" the timezone and maybe that's why this is throttled, but not really focused on improving this particular feature.

File: src/libs/actions/Report.js
Comment (marcaaron) [reply src/libs/actions/Report.js:964]: Don't really have strong feelings one way or the other on this.

File: tests/README.md
Comment (marcaaron) [reply tests/README.md:72]: yes good catch.

**Comment by Julesssss** [comment]
Seems like this still hasn't hit prod yet? As NewDot deploy is blocked by this issue could you please[ comment on this issue](https://github.com/Expensify/App/issues/9586) when it gets deployed to prod? Thank you!

**Comment by kavimuru** [comment]
@marcaaron We never get notification in mweb. 
Section 2:

>  1. Testing a report comment browser notification appears when the chat is not active (web)
>  2.  Create two accounts and a chat between them. Open a new NewDot window in a new Chrome session (do not use incognito mode).
>  3.  Send message from Account A to Account B while Account B is focused on another chat
>  4.  Verify a browser notification appears for Account B
>

So step 4, No notification received in mWeb environment.

**Comment by yuwenmemon** [comment]
Yeah @kavimuru that isn't a blocker :+1:

**Pull request merged**
